Linked list with void pointers.c

```diff
@@ -37,37 +32,65 @@ void display_float(node *tail)
 {
     printf("%lf\n", *(double *)tail->data);
 }
-void display(void(*display_datatype)(node *tail), node *tail)
+void display_string(node *tail)
+{
+    printf("%c\n", *(char *)tail->data);
+}
+void display( node *tail)
 {
-    if (head == NULL)
+    if (tail == NULL)
         printf("List is empty\n");
     else
     {
         while (tail)
         {
-            display_datatype(tail);
+            tail->funptr(tail);
             tail = tail->next;
         }
     }
 }
+
+int get_size_int(int x)
+{
+    return sizeof(int);
+}
+int get_size_float(double x)
+{
+    return sizeof(double);
+}
+int get_size_char(char x)
+{
+    return sizeof(char);
+}
+int get_size(int (*function)(void *ptr), void *ptr)
+{
+    return function(ptr);
+}
 int main()
 {
-    int intnum = 0;
+    note note;
+    note.head=NULL;
+    note.tail=NULL;
     int i;
+    int intnum = 0;
     for (i = 0; i<10; i++)
     {
         intnum++;
-        insert(&head, &tail, (const void *)&intnum, sizeof(intnum));
+        insert(&note.tail,display_int, (const void *)&intnum,get_size(get_size_int,(void *)&intnum));
+        if( note.head==NULL)

```

Comment:
write functions for insert_int, insert_float, insert_char, insert_string with one parameter i.e (datatype) data, then call insertnode within the respective functions

Reply:
ok